traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -699,62 +759,301 @@ WHERE
 }
 
 func updateV15(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, reqDS *tc.DeliveryServiceNullableV15) (*tc.DeliveryServiceNullableV15, int, error, error) {
```

Comment:
So this function should actually stay exactly the same as it was before. Basically, we want to rename `updateV30` to `updateV31`, change it to accept and return V31 structs, then copy `updateV15` to `updateV30`, change it to accept V30 structs, upgrade them to V31 by changing the query to only select the new V31 field from the DB, then pass it to `updateV31` for the actual handling.

It's kind of hard to draw in a review, but the functions are designed to chain together like so:
```
UpdateV12 -> updateV12
              |     ^
              V     |
UpdateV13 -> updateV13
              |     ^
              V     |
   ...          ...
UpdateV31 -> updateV31  <------(this latest handler does all the actual handling)
```
So each API version is like a link in a chain. For adding a new version, you just add a new link to the end of the chain, and you don't have to worry about earlier links in the chain (other than the one you're linking to). Does that make sense?